src/editor/gltf/GLTFLoader.js

```diff
@@ -1990,6 +1995,22 @@ class GLTFLoader {
       ) {
         object.userData.gltfExtensions = object.userData.gltfExtensions || {};
         object.userData.gltfExtensions[name] = objectDef.extensions[name];
+        if (name === "MOZ_hubs_components") {
+          const components = object.userData.gltfExtensions[name];
+          for (const [componentName, componentProps] of Object.entries(components)) {
+            for (const [propName, propValue] of Object.entries(componentProps)) {
+              if (
+                HUBS_NODEREF_COMPONENTS[componentName] &&
+                HUBS_NODEREF_COMPONENTS[componentName].indexOf(propName) !== -1
+              ) {
+                this.getDependency("node", propValue).then(node => {
+                  node.userData.MOZ_spoke_uuid = node.uuid;
+                  componentProps[propName] = { __gltfIndexForUUID: node.uuid };
+                });
```

Comment:
`getDependency()` is async method so no guarantee that it is resolved by the end of loader's parse (= by loader's `onLoad()` callback is called). I'm not familiar with Spoke yet but I think it would be problematic, wouldn't it?

Reply:
Yeah its definitely not ideal. Doing this the "right" way would mean refactoring GLTF loader some more and since we are going to replace it soon it doesn't seem worth it. This info is only used on export so the promise is going to be well resolved by then.

Reply:
Given how we're loading glTFs, loading and awaiting the whole scene, I think it'll be fine. If you're able to get this to turn into a race condition, you're going to have other problems. Like the model just isn't in the output.

Reply:
I would like to suggest to add a comment here about why we don't need to wait. Otherwise, other devs might be confused whey they read this part.

Reply:
Missed this before pushing the PR, but added a comment now.